### ISSUE TITLE
A tutorial of using this repo

### ISSUE THREAD

**Issue by GabrielLin** [Farahn/AES#5]
Thanks for your great work. Could you please provide a tutorial of using this repo to reproduce the results in your paper. Where to start? Really thanks.

**Comment by Farahn** [comment]
I can update in more detail later if needed, here is a short set of steps. For the two data sets, ASAP and TOEFL (LDC), the first step is to run the data parse scripts, either ASAP_dataparse.ipynb or TOEFL_dataparse.ipynb. After that the training or testing scripts can be run for all models except the ones that use BERT. For the models using BERT, the BERT preprocessing scripts should be run before the training or testing (BERT_text_representation.ipynb). 

Assuming that you will be using the discourse marker BCA model (DM-BCA) for the LDC TOEFL data, here are the steps:
- The LDC data folder has three splits for train, test and dev stored in files index-train.csv, index-test.csv and index-dev.csv . 
- Running the script in TOEFL_dataparse.ipynb will preprocess and prepare the data. 
- The pretrained model can be downloaded at https://sites.google.com/site/nadeemf0755/research/automatic-essay-scoring
- Running the notebook  DM_BCA_test.ipynb will use the pretrained model (bca_dm_toefl_model) and the preprocessed data, and will load and run the model on the test set. This will reproduce the results in the paper. 
-The script DM_BCA_train.ipynb can be used to train the model from scratch after preprocessing the data. 

Let me know if this is helpful, I am happy to add more details and answer specific questions.

**Comment by GabrielLin** [comment]
If I do not have the LDC data, which test can I do with the pre-trained models? Thanks.

**Comment by Farahn** [comment]
You can use the Kaggle data set, ASAP (https://www.kaggle.com/c/asap-aes). The steps for analysis would be the same, using the ASAP* scripts instead of TOEFL* scripts.

**Comment by GabrielLin** [comment]
I have sucessfully apply LDC2014T06 from LDC. I can study all your code now. Happy!
